tools/gas_cost_measures.py

```diff
@@ -0,0 +1,220 @@
+from binascii import hexlify
+from os import urandom
+
+from coincurve import PrivateKey
+from eth_tester import EthereumTester, PyEVMBackend
+from web3 import EthereumTesterProvider, Web3
+
+from raiden.transfer.balance_proof import pack_signing_data
+from raiden.transfer.utils import hash_balance_data
+from raiden.encoding.signing import sign
+
+from raiden_contracts.contract_manager import ContractManager, CONTRACTS_SOURCE_DIRS
+from raiden_contracts.tests.fixtures import contracts_manager  # noqa
+from raiden_contracts.tests.utils import get_pending_transfers_tree
+# TODO Remove the unused import when possible.
+# We need it right now to avoid a circular import inside raiden-contracts.
+
+# increase block gas limit
+import eth_tester.backends.pyevm.main as pyevm_main
+pyevm_main.GENESIS_GAS_LIMIT = 6 * 10 ** 6
+
+
+class ContractTester:
```

Comment:
Why not add this to the build? We already have tools in place to test the smart contracts:

- [fixture to instantiate a tester](https://github.com/raiden-network/raiden/blob/d625611a4fee218e68ce2de7d277979f0450f4de/raiden/tests/integration/fixtures/blockchain.py#L33-L41)
- [number of raiden nodes in the network](https://github.com/raiden-network/raiden/blob/d625611a4fee218e68ce2de7d277979f0450f4de/raiden/tests/fixtures/variables.py#L111-L114) which equals to the number of accounts in the tester
- [default balance of each account](https://github.com/raiden-network/raiden/blob/d625611a4fee218e68ce2de7d277979f0450f4de/raiden/tests/fixtures/variables.py#L91-L96)
- [the fixture which funds the accounts](https://github.com/raiden-network/raiden/blob/master/raiden/tests/integration/fixtures/blockchain.py#L113)
- [fixtures to deploy the required smart contracts](https://github.com/raiden-network/raiden/blob/d625611a4fee218e68ce2de7d277979f0450f4de/raiden/tests/integration/fixtures/smartcontracts.py)

The only detail is that this should be skipped if the blockchain is of type geth **or** the type **tester** should be forced.

Reply:
These fixtures depend on multiple others, so using them in a standalone script outside pytest would get ugly. My options were thus either to duplicate this functionality and write a clean standalone tool like this, or to hack the computing of `MAXIMUM_PENDING_TRANSFERS` into a test somehow (parametrizing a test with multiple Merkle tree widths, assert the gas limit is kept and see at which value it fails?) I thought this is the better solution. I can of course add a test to the build that the given `MAXIMUM_PENDING_TRANSFERS` value is sane and then reuse it for a computing tool that works inside pytest.

Reply:
> These fixtures depend on multiple others, so using them in a standalone script outside pytest would get ugly

Yeah, I was not suggesting to implement a standalone tool, but to write a test.